actors/builtin/multisig/multisig_test.go

```diff
@@ -311,6 +311,519 @@ func TestApprove(t *testing.T) {
+			Approved: []addr.Address{anne},
+		})
+	})
+
+	t.Run("fail to cancel transaction when not signer", func(t *testing.T) {
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, numApprovals, noUnlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		serParams := serde.MustSerialize(&multisig.ChangeNumApprovalsThresholdParams{NewThreshold: 1000})
+		actor.propose(rt, chuck, sendValue, builtin.Method_MultiSigActor_ChangeNumApprovalsThreshold, serParams)
+		rt.Verify()
+
+		rt.SetBalance(sendValue)
+		rt.SetCaller(richard, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		rt.ExpectAbort(exitcode.ErrForbidden, func() {
+			actor.cancel(rt, txnID)
+		})
+		rt.Verify()
+
+		// Transaction should remain after invalid cancel
+		actor.assertTransactions(rt, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    sendValue,
+			Method:   builtin.Method_MultiSigActor_ChangeNumApprovalsThreshold,
+			Params:   serParams,
+			Approved: []addr.Address{anne},
+		})
+	})
+
+	t.Run("fail to cancel a transaction that does not exist", func(t *testing.T) {
+		rt := builder.Build()
+		const dneTxnID = int64(1)
+
+		actor.constructAndVerify(rt, numApprovals, noUnlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		serParams := serde.MustSerialize(&multisig.ChangeNumApprovalsThresholdParams{NewThreshold: 1000})
+		actor.propose(rt, chuck, sendValue, builtin.Method_MultiSigActor_ChangeNumApprovalsThreshold, serParams)
+		rt.Verify()
+
+		rt.SetBalance(sendValue)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		rt.ExpectAbort(exitcode.ErrIllegalState, func() {
+			actor.cancel(rt, dneTxnID)
+		})
+		rt.Verify()
+
+		// Transaction should remain after invalid cancel
+		actor.assertTransactions(rt, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    sendValue,
+			Method:   builtin.Method_MultiSigActor_ChangeNumApprovalsThreshold,
+			Params:   serParams,
+			Approved: []addr.Address{anne},
+		})
+	})
+}
+
+func TestAddSigner(t *testing.T) {
+	actor := msActorHarness{multisig.MultiSigActor{}, t}
+
+	receiver := newIDAddr(t, 100)
+	anne := newIDAddr(t, 101)
+	bob := newIDAddr(t, 102)
+	chuck := newIDAddr(t, 103)
+
+	const noUnlockDuration = int64(0)
+	const numApprovals = int64(3)
+	var initialSigner = []addr.Address{anne, bob}
+	var addedSigner = chuck
+
+	builder := mock.NewBuilder(context.Background(), t, receiver).WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple add signer", func(t *testing.T) {
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, numApprovals, noUnlockDuration, initialSigner...)
+
+		rt.SetCaller(receiver, builtin.AccountActorCodeID)
```

Comment:
Please test that if the receiver is not the caller, the call is rejected.

Reply:
Talked in person, going to skip this as that would only test the mockrt.